views/layout/movieRender.ejs

```diff
@@ -1,15 +1,13 @@
-        <% if (Array.isArray(movieRender)) { %>
-          <% movieRender.forEach(movie => { %>
-        <div class="movieTemplate__card">
-          <div class="movieTemplate__cover" style="background-image:url(<%= movie.image.url%>)"></div>
-          <div class="movieTemplate__img" style="background-image:url(<%= movie.image.url%>)"></div>
-          <h2 class="movieTemplate__title"><%= movie.title  %></h2>
-          <a class="movieTemplate__button" href="/movies/<%- movie.id %>">
-            Se mer
+        
+        <% if (typeof movieRender === 'string' || movieRender instanceof String ){ %>
```

Comment:
Here's that complex logic that is required because you're using the same template file for all three outcomes. Having a lot of logic in your templates makes them complicated. Remember the principle of separation of concerns.

It would have been better if all of the logic to decide WHAT to render (movies list, movie page or 404) stayed in the route, and the templates only had to care about HOW to render, i.e. what it should look like.